policies/rules/security/entry.go

```diff
@@ -0,0 +1,750 @@
+	SourceDevices        *util.MemberType `xml:"source-hip"`
+	DestinationDevices   *util.MemberType `xml:"destination-hip"`
+	Schedule             *string          `xml:"schedule,omitempty"`
+	Tags                 *util.MemberType `xml:"tag"`
+	NegateSource         *string          `xml:"negate-source,omitempty"`      // bool
+	NegateDestination    *string          `xml:"negate-destination,omitempty"` // bool
+	Disabled             *string          `xml:"disabled,omitempty"`           // bool
+	Description          *string          `xml:"description,omitempty"`
+	GroupTag             *string          `xml:"group-tag,omitempty"`
+	Action               string           `xml:"action"`
+	IcmpUnreachable      *string          `xml:"icmp-unreachable,omitempty"` // bool
+	Type                 *string          `xml:"rule-type,omitempty"`
+	Options              *secOptions      `xml:"option"`
+	LogSetting           *string          `xml:"log-setting,omitempty"`
+	LogStart             *string          `xml:"log-start,omitempty"` // bool
+	LogEnd               *string          `xml:"log-end,omitempty"`   // bool, default is true
+	ProfileSettings      *profileSettings `xml:"profile-setting,omitempty"`
+	Qos                  *qos             `xml:"qos,omitempty"`
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+func (e *Entry1) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	type local Entry1
+	yes := util.YesNo(true)
+	ans := local{
+		LogEnd: &yes,
+	}
+	if err := d.DecodeElement(&ans, &start); err != nil {
+		return err
+	}
+	*e = Entry1(ans)
+	return nil
+}
+
+type secOptions struct {
+	DisableServerResponseInspection *string `xml:"disable-server-response-inspection,omitempty"` // bool
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+type profileSettings struct { // One of.
+	Groups   *util.MemberType         `xml:"group,omitempty"`
+	Profiles *profileSettingsProfiles `xml:"profiles,omitempty"`
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+type profileSettingsProfiles struct {
+	UrlFilteringProfiles     *util.MemberType `xml:"url-filtering,omitempty"`
+	DataFilteringProfiles    *util.MemberType `xml:"data-filtering,omitempty"`
+	FileBlockingProfiles     *util.MemberType `xml:"file-blocking,omitempty"`
+	WildfireAnalysisProfiles *util.MemberType `xml:"wildfire-analysis,omitempty"`
+	AntiVirusProfiles        *util.MemberType `xml:"virus,omitempty"`
+	AntiSpywareProfiles      *util.MemberType `xml:"spyware,omitempty"`
+	VulnerabilityProfiles    *util.MemberType `xml:"vulnerability,omitempty"`
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+type qos struct {
+	Marking *qosMarking `xml:"marking,omitempty"`
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+type qosMarking struct { // One of.
+	IpDscp                   *string `xml:"ip-dscp,omitempty"`
+	IpPrecedence             *string `xml:"ip-precedence,omitempty"`
+	FollowClientToServerFlow any     `xml:"follow-c2s-flow,omitempty"` // exist bool
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+func SpecMatches(a, b *Entry) bool {
```

Comment:
I have strong feeling that this function can be simplify as well.

Reply:
We discussed a bit in Friday's meeting.  I also recommend pursuing this once we are farther along on the generator as a whole.